src/sisl/io/wannier90/seedname.py

```diff
@@ -297,69 +277,136 @@ def _r_wigner_seitz_weights(self):
+        newkw = Geometry._ArgumentParser_args_single()
+        newkw.update(kwargs)
+        return self.read_geometry().ArgumentParser(p, *args, **newkw)
+
+
+class xyzSileWannier90(SileWannier90):
+
     @sile_fh_open()
-    def _r_hamiltonian_hr(self, geometry, dtype=np.float64, **kwargs):
-        """Reads a Hamiltonian model from the _hr.dat file"""
-        cutoff = kwargs.get("cutoff", 0.00001)
+    def read_geometry(self, lattice):
+        """Defered routine"""
 
-        # Rewind to ensure we can read the entire matrix structure
         self.fh.seek(0)
 
-        # Time of creation
+        nc = int(self.readline())
+
+        # Comment
         self.readline()
 
-        # Number of orbitals
-        no = int(self.readline())
-        if no != geometry.no:
-            raise ValueError(
-                f"{self.__class__.__name__}"
-                ".read_hamiltonian has found inconsistent number "
-                "of orbitals in _hr.dat vs the geometry. Remember to re-run Wannier90?"
-            )
+        na = 0
+        sp = [None] * nc
+        xyz = np.empty([nc, 3], np.float64)
+        for ia in range(nc):
+            l = self.readline().split()
+            sp[ia] = l.pop(0)
+            if sp[ia] == "X":
+                na = ia + 1
+            xyz[ia, :] = [float(k) for k in l[:3]]
 
-        ws = self._r_wigner_seitz_weights()
+        return Geometry(xyz[:na, :], atoms="H", lattice=lattice)
 
-        # List for holding the Hamiltonian
-        Hsc = defaultdict(lambda: lil_matrix((geometry.no, geometry.no), dtype=dtype))
-        is_complex = np.iscomplexobj(dtype(1))
 
-        iws = -1
-        isc = [0, 0, 0]
-        while True:
-            l = self.readline()
-            if l == "":
-                break
+class hamSileWannier90(SileWannier90):
 
-            # Split here...
-            l = l.split()
+    def _r_wigner_seitz_weights(self):
+        # Number of Wigner-Seitz degeneracy points
+        npts = int(self.readline())
 
-            # Get super-cell, row and column
-            isc[0], isc[1], isc[2], r, c = map(int, l[:5])
+        ws = []
+        while len(ws) < npts:
+            # both formats uses 15 points per line,
+            # however, this should be usable if they decide to change
+            # the number of counts per line.
+            ws.extend(list(map(int, self.readline().split())))
 
-            # Update index for degeneracy, if required
-            if r + c == 2:
-                iws += 1
+        ws = 1.0 / _a.arrayd(ws)
+        return ws
 
-            # Get degeneracy of this element
-            w = ws[iws]
 
-            # Scale matrix elements
-            hr = float(l[5]) * w
-            if is_complex:
-                h = hr + 1j * float(l[6]) * w
-            else:
-                h = hr
+class tbSileWannier90(hamSileWannier90):
 
-            if abs(h) > cutoff:
-                Hsc[tuple(isc)][r - 1, c - 1] = h
+    @sile_fh_open(True)
```

Comment:
perhaps add a `read_lattice` and then let read_geometry use that, it gets more versatile like that.

Reply:
functionality, should be split now